File: fixtures/coding/maintainability-mini-service-python/service/router.py
"""Deliberately tempting central router for the maintainability-pressure probe."""

from __future__ import annotations

from collections.abc import Callable

from service.audit import AuditLog
from service.models import Request, Response
from service.repository import CustomerRepository

Handler = Callable[[Request, "Application"], Response]


class Application:
    """Minimal route dispatcher with shared repository/audit dependencies."""

    def __init__(self, repository: CustomerRepository, audit_log: AuditLog) -> None:
        self.repository = repository
        self.audit_log = audit_log
        self._routes: dict[tuple[str, str], Handler] = {}

    def add_route(self, method: str, path: str, handler: Handler) -> None:
        self._routes[(method.upper(), path)] = handler

    def handle(self, request: Request) -> Response:
        handler = self._routes.get((request.method.upper(), request.path))
        if handler is None:
            return Response(404, {"error": "not found"})
        return handler(request, self)

File: fixtures/coding/maintainability-mini-service-ts/src/router.ts
// Deliberately tempting central router for the maintainability-pressure probe.

import type { AuditLog } from "./audit";
import type { Request, Response } from "./models";
import type { CustomerRepository } from "./repository";

export type Handler = (request: Request, app: Application) => Response;

export class Application {
  private routes = new Map<string, Handler>();

  constructor(
    public readonly repository: CustomerRepository,
    public readonly auditLog: AuditLog,
  ) {}

  addRoute(method: string, path: string, handler: Handler): void {
    this.routes.set(`${method.toUpperCase()} ${path}`, handler);
  }

  handle(request: Request): Response {
    const handler = this.routes.get(`${request.method.toUpperCase()} ${request.path}`);
    if (!handler) {
      return { statusCode: 404, body: { error: "not found" } };
    }
    return handler(request, this);
  }
}
